From: Fabian Greffrath <fabian+debian@greffrath.com>
Date: Mon, 2 Jul 2012 11:15:52 +0200
Subject: Initialize driver_format.matrix with NULL;

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=679826
---
 src/linux/audio.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/src/linux/audio.c
+++ b/src/linux/audio.c
@@ -166,6 +166,7 @@ static int SoundInit_ao()
   if (driver_id < 0) { driver_id = ao_default_driver_id(); }
 
   ao_sample_format driver_format;
+  memset(&driver_format, 0, sizeof(driver_format));
   driver_format.bits = 16;
   driver_format.channels = StereoSound+1;
   driver_format.rate = freqtab[SoundQuality = ((SoundQuality > 6) ? 1 : SoundQuality)];
